gtsam/slam/ProjectionFactor.h

```diff
@@ -99,6 +99,17 @@ namespace gtsam {
     /** Virtual destructor */
     ~GenericProjectionFactor() override {}
 
+    /// @return the measured pixel coordinates
+    auto measured() const -> const Point2 & { return measured_; }
+
+    /// @return the camera calibration
+    auto K() const -> const boost::shared_ptr<CALIBRATION> & { return K_; }
```

Comment:
This is interesting syntax I've never seen before. Is this some new convention or style?

Reply:
@varunagrawal I think it was introduced in C++11... it's useful when the return type on the left is too long and makes hard reading all function names. A matter of style, mostly, of course ;-)

Reply:
[Here](https://en.cppreference.com/w/cpp/language/function) you can see a more proper definition. Search for "trailing return type".

Reply:
This is amazing! I was able to use this immediately in another PR, thanks @jlblancoc  :smile: